chanzuckerberg/corpora-data-portal
# thuang-517-AuthN

When the user is logged in, you see the random scientist emoji:

<img width="1218" alt="Screen Shot 2020-11-12 at 2 06 03 PM" src="https://user-images.githubusercontent.com/6309723/99001836-5f075880-24f0-11eb-888b-ad4faceae8fa.png">

When a user is logged in and clicks on the emoji:
<img width="465" alt="Screen Shot 2020-11-12 at 2 06 12 PM" src="https://user-images.githubusercontent.com/6309723/99001845-629adf80-24f0-11eb-8286-8308634af830.png">

UPDATE (11/16): Use email first char if name is not available

![demo3](https://user-images.githubusercontent.com/6309723/99344085-1deeab80-2844-11eb-9941-748487c2fa54.gif)
![demo2](https://user-images.githubusercontent.com/6309723/99344098-25ae5000-2844-11eb-8ea4-92a780964390.gif)

When user is not logged in, but they click on "Create Collection" button:
<img width="1216" alt="Screen Shot 2020-11-12 at 2 05 44 PM" src="https://user-images.githubusercontent.com/6309723/99001813-57e04a80-24f0-11eb-9f0b-15d996beab10.png">

(UPDATED: 11/13) When a user is not logged in and `auth` flag is `true`, they see "Log In" button:
<img width="1462" alt="Screen Shot 2020-11-13 at 11 20 42 AM" src="https://user-images.githubusercontent.com/6309723/99112404-7868dd00-25a2-11eb-9a14-cabe28b1203d.png">

GIF:
![demo](https://user-images.githubusercontent.com/6309723/99003154-76474580-24f2-11eb-8a86-64a8e0bd395f.gif)

QA Steps:
1. Pull this branch: `thuang-517-AuthN`
1. Run `npm start`
1. Navigate to: `http://localhost:8000/?auth=true&cc=true` to enable both flags
1. You should now see "Log In" and "Sign Up" buttons, and if you click on "Create Collection" button, you'll get the CTA
1. Now to simulate a logged in user's experience, I use [Requestly](https://chrome.google.com/webstore/detail/requestly-redirect-url-mo/mdnleldcmiljblolnjhpnblkcekpdkpa?hl=en) to modify the request header's `Cookie` to include my `dev` env cookies. Essentially you log in on dev cellxgene (https://api.cellxgene.dev.single-cell.czi.technology/dp/v1/login), and then inspect a request we send to BE, so you can cookie the request header `Cookie` field and paste that into Requestly rule, with `contains URL: api.cellxgene.dev.single-cell.czi.technology`. Press "Save"

    E.g., 

    **NOTE: Please check your `frontend/src/configs/configs.js` uses `API_URL: "https://api.cellxgene.dev.single-cell.czi.technology"` instead of the old `corpora` namespace**

    <img width="916" alt="Screen Shot 2020-11-12 at 2 29 35 PM" src="https://user-images.githubusercontent.com/6309723/99004068-86135980-24f3-11eb-9c66-2e742396133f.png">

1. Now activate Requestly like so:
    
    <img width="683" alt="Screen Shot 2020-11-12 at 2 34 30 PM" src="https://user-images.githubusercontent.com/6309723/99004597-3c773e80-24f4-11eb-92b6-e3f7436ec27e.png">

1. Refresh the page, and you should see all the logged in features!

1. Alternatively, you can also run local BE with `DEPLOYMENT_STAGE=test`! 

PTAL thanks all!

resolves #571
